src/input_context/bind.rs

```diff
@@ -0,0 +1,164 @@
+
+#[derive(Debug)]
+pub struct BindConfig {
+    pub input: Input,
+    pub info: BindConfigInfo,
+    pub ignored: bool,
+}
+
+impl BindConfig {
+    pub fn new(input: impl Into<Input>) -> Self {
+        Self {
+            input: input.into(),
+            ignored: true,
+            info: BindConfigInfo::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BindConfigs {
+    One(BindConfig),
+    Set(BindConfigSet),
+}
+
+impl BindConfigs {
+    pub fn info(&self) -> &BindConfigInfo {
+        match self {
+            Self::One(config) => &config.info,
+            Self::Set(set) => &set.info,
+        }
+    }
+
+    pub fn info_mut(&mut self) -> &mut BindConfigInfo {
+        match self {
+            Self::One(config) => &mut config.info,
+            Self::Set(set) => &mut set.info,
+        }
+    }
+}
+
+pub trait IntoBindConfigs
+where
+    Self: Sized,
+{
+    fn into_configs(self) -> BindConfigs;
+
+    fn with_modifier(self, modifier: impl InputModifier) -> BindConfigs {
+        self.into_configs().with_modifier(modifier)
+    }
+
+    fn with_condition(self, condition: impl InputCondition) -> BindConfigs {
+        self.into_configs().with_condition(condition)
+    }
+}
+
+impl IntoBindConfigs for BindConfigs {
+    fn into_configs(self) -> BindConfigs {
+        self
+    }
+
+    fn with_modifier(mut self, modifier: impl InputModifier) -> BindConfigs {
+        self.info_mut().modifiers.push(Box::new(modifier));
+        self
+    }
+
+    fn with_condition(mut self, condition: impl InputCondition) -> BindConfigs {
+        self.info_mut().conditions.push(Box::new(condition));
+        self
+    }
+}
+
+impl IntoBindConfigs for BindConfig {
+    fn into_configs(self) -> BindConfigs {
+        BindConfigs::One(self)
+    }
+}
+
+impl<T: Into<Input>> IntoBindConfigs for T {
+    fn into_configs(self) -> BindConfigs {
+        BindConfig::new(self).into_configs()
+    }
+}
+
+impl IntoBindConfigs for BindConfigSet {
+    fn into_configs(self) -> BindConfigs {
+        BindConfigs::Set(self)
+    }
+}
+
+impl<T: IntoBindConfigs, const N: usize> IntoBindConfigs for [T; N] {
+    fn into_configs(self) -> BindConfigs {
+        BindConfigSet::from_iter(self.into_iter().map(IntoBindConfigs::into_configs)).into_configs()
+    }
+}
+
+impl IntoBindConfigs for () {
+    fn into_configs(self) -> BindConfigs {
+        BindConfigSet::default().into_configs()
+    }
+}
+
+impl<T: IntoBindConfigs> IntoBindConfigs for (T,) {
+    fn into_configs(self) -> BindConfigs {
+        self.0.into_configs()
+    }
+}
+
+macro_rules! impl_tuple {
+    ($(#[$meta:meta])* $($T:ident $i:tt),*) => {
+        $(#[$meta])*
+        impl<$($T: IntoBindConfigs),*> IntoBindConfigs for ($($T,)*) {
+            fn into_configs(self) -> BindConfigs {
+                BindConfigSet::from_iter([
+                    $(self.$i.into_configs()),*
+                ])
+                .into_configs()
+            }
+        }
+    };
+}
+
+impl_tuple!(A 0, B 1);
```

Comment:
Bevy already provides a helper for it: https://docs.rs/bevy/latest/bevy/utils/macro.all_tuples.html
Usage example from replicon: https://github.com/projectharmonia/bevy_replicon/blob/df44894205625e6e3dd8f627d6651545d12d3030/src/core/replication/replication_rules.rs#L329

Reply:
I tried using this, but it doesn't work here because we need access to the index of the field we're using as well, i.e. in `(T0, T1)` we need `0, 1`. So I had to implement this manually.

Reply:
Ah, I see!